spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -123,7 +124,222 @@ In Jakarta Data, an entity refers to a fundamental data representation and manag
+
+Example:
+
+[source,java]
+----
+public class Address {
+
+    private String street;
+    private String city;
+    private String postalCode;
+}
+
+//the entity
+public class Person {
+    private Long id;
+
+    private String name;
+    private Address address;  // This is a component field
+}
+----
+
+The structure of a Document, wide-column, and Graph will be like the JSON representation below. The JSON representation is for illustration; it belongs to the NoSQL database and defines the final serialization structure such as JSON and BSON, including another field such as timestamp for clock-vector, revision, and other internal purposes. The JSON and the following SQL table are representations resulting from the Java classes.
+
+[source,json]
+----
+{
+  "id": 1,
+  "name": "John Doe",
+  "street": "123 Main St",
+  "city": "Sampleville",
+  "postalCode": "12345"
+}
+----
+
+[options="header"]
+|====
+| id | name     | street        | city
+| 1  | John Doe | 123 Main St   | Sampleville
+|====
+
+This approach allows for a flat and denormalized structure in the database, making it suitable for certain NoSQL database types.
+
+2. * Second Approach: Storing Components in Separate Tables (Relational Databases) or as Subdocuments/UDTs (NoSQL Databases)*
+
+This approach is typically employed for more complex relationships and associations within the domain model, allowing for greater flexibility and scalability. It involves storing components, such as the `Passport` in the example, in separate tables for relational databases or as subdocuments or User-Defined Types (UDTs) for NoSQL databases. This method is suitable for scenarios where an association exists between one entity and another.
+
+For instance, consider the `Citizen` and `Passport` classes. In a relational database, this approach results in two separate tables, each representing an entity with its associated persistence context. In contrast, for a NoSQL database like a Document database, the `Passport` can still act as a component within the `Citizen` entity, signifying that a `Passport` is closely tied to a `Citizen`. However, the modeling may vary depending on the specific NoSQL database and its capabilities.
+
+
+[source,java]
+----
+public class Passport {
+    private Long id;
+    private String passportNumber;
+    private LocalDate expirationDate;
+}
+
+public class Citizen {
+    private Long id;
+    private String name;
+    private Passport passport; // One-to-One relationship with Passport for relational database
+}
+----
+
+Here's a representation of the `Citizen` and `Passport` classes in both table and JSON formats:
+
+[options="header"]
+|===
+| Citizen Table | Passport Table
+| id (Primary Key) | id (Primary Key)
+| name | passportNumber
+| passport_id (Foreign Key) | expirationDate
+|===
```

Comment:
It is confusing to show 2 database tables in a single table.  I would expect to see two tables for this, one for each, and each including the same example data that you show for the JSON example.